### PR TITLE
rust: format rust code using rustfmt

### PR DESCRIPTION
**Pull request by isphinx**
rust: add auto format feature

### CODE REVIEW

File: layers/+lang/rust/README.org [hunk 1]
```diff
@@ -33,6 +34,22 @@ To enable auto-completion, ensure that the =auto-completion= layer is enabled.
 [[http://doc.crates.io/index.html][Cargo]] is a project management command line tool for Rust. Installation
 instructions can be found on the main page of [[http://doc.crates.io/index.html][Cargo]].
 
+** Rustfmt
+
+To formatting Rust code according to style guidelines using [[https://github.com/rust-lang-nursery/rustfmt][rustfmt]].
```
Comment (fbergroth) [inline layers/+lang/rust/README.org]: I think it should read:
> Format Rust code ...

(no need to close this PR, just force-push to this branch)

File: layers/+lang/rust/README.org [hunk 2]
```diff
@@ -33,6 +34,22 @@ To enable auto-completion, ensure that the =auto-completion= layer is enabled.
...
+
+#+BEGIN_SRC sh
+cargo install rustfmt
+#+END_SRC
+
+To enable automatic buffer formatting on save
```
Comment (fbergroth) [inline layers/+lang/rust/README.org]: > To enable automatic buffer formatting on save, set the variable =rust-enable-rustfmt-on-save= to =t=.

File: layers/+lang/rust/packages.el [hunk 3]
```diff
@@ -72,3 +73,16 @@
       (spacemacs/declare-prefix-for-mode 'rust-mode "mg" "goto")
       (spacemacs/set-leader-keys-for-major-mode 'rust-mode
         "gg" 'racer-find-definition))))
+
+(defun rust/init-rustfmt ()
+  (use-package rustfmt
+    :defer t
+    :init
+    (progn
+      (if rust-enable-rustfmt-on-save
```
Comment (fbergroth) [inline layers/+lang/rust/packages.el]: Use `when` instead of `if` when there is no else-branch.

File: layers/+lang/rust/packages.el [hunk 4]
```diff
@@ -72,3 +73,16 @@
...
+          (spacemacs/add-to-hook 'rust-mode-hook
+                                 '(rustfmt-enable-on-save)))
+
+      (spacemacs/set-leader-keys-for-major-mode 'rust-mode
+        "=" 'rustfmt-format-buffer))
+    ))
```
Comment (fbergroth) [inline layers/+lang/rust/packages.el]: No, whitespace between trailing `)`. Format it like this: `"=" 'rustfmt-format-buffer))))`

**Comment by isphinx** [comment]
@fbergroth Thank you for your patience advice👍

**Comment by fbergroth** [comment]
:+1:

**Comment by syl20bnr** [comment]
Thank you :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**